cli/loader_lib.c

```diff
@@ -161,11 +324,42 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
     }
 
     // Introspect to find our own path
-    const char * lib_dir = jl_get_libdir();
+    const char *lib_dir = jl_get_libdir();
 
     // Pre-load libraries that libjulia-internal needs.
     int deps_len = strlen(dep_libs);
-    char * curr_dep = &dep_libs[0];
+    char *curr_dep = &dep_libs[0];
+
+    void *cxx_handle;
+    int done_probe = 0;
+#if defined(_OS_LINUX_)
+    int do_probe = 1;
+    char *probevar = getenv("JULIA_PROBE_LIBSTDCXX");
+    if (probevar) {
+        if (strcmp(probevar, "1") == 0 || strcmp(probevar, "yes"))
+            do_probe = 1;
+        else if (strcmp(probevar, "0") == 0 || strcmp(probevar, "no"))
+            do_probe = 0;
+    }
+    if (do_probe) {
+        char *cxxpath = libstdcxxprobe();
+        if (cxxpath) {
+            cxx_handle = dlopen(cxxpath, RTLD_LAZY);
+            char *dlr = dlerror();
+            if (dlr) {
+                jl_loader_print_stderr("ERROR: Unable to dlopen(cxxpath) in parent!\n");
+                jl_loader_print_stderr3("Message: ", dlerr, "\n");
+                exit(1);
+            }
+            free(cxxpath);
+            done_probe = 1;
+        }
+    }
+#endif
+    // If not on linux, or the probe does not finish successfully, load the bundled version.
+    if (!done_probe) {
+        load_library("libstdc++.so.6", lib_dir, 1);
```

Comment:
On third thought, I think the best thing to do here is to disambiguate the "not on linux" and "probe did not finish successfully" cases; on e.g. macOS, there's no need to eagerly load `libstdc++`, so let's just not do it.  It will just get loaded on-demand, along with everything else.  On Linux, if our probe failed, we should load the bundled version.

Reply:
This will also fix Windows CI, since you're trying to load `libstdc++.so.6` on windows, which doesn't use `.so`. ;)